demo-app/client/src/tests/post/order.service.spec.ts

```diff
@@ -21,21 +21,38 @@ describe(`OrderService`, () => {
 
   it(`should check request parameters are correct`,
     async(
-
       inject([OrderService, HttpTestingController],
         (service: OrderService, backend: HttpTestingController) => {
 
-        service.order({orderDto: {id: 15, name: 'example order'}, producer: 'test-producer'}).subscribe();
-
+        const bodyParam = {id: 15, name: 'example order'};
+        service.order({orderDto: bodyParam, producer: 'test-producer'}).subscribe();
         backend.expectOne((req: HttpRequest<any>) => {
-          return req.method === 'POST'
-            && req.url === '/api/order'
-            && req.body.id === 15
-            && req.body.name === 'example order'
-            && req.params.toString() === 'producer=test-producer';
+          expect(req.method).toEqual('POST');
```

Comment:
I believe this construct is more informative on failure, and recommend its usage. Compare:

**original**

```
OrderService should check request parameters are correct FAILED
	Failed: Expected one matching request for criteria "Match by function: ", found none.
	Error: Expected one matching request for criteria "Match by function: ", found none.
	    at HttpClientTestingBackend.webpackJsonp.../../../common/esm5/http/testing.js.HttpClientTestingBackend.expectOne Users/john/dev/swagger-angular-generator/demo-app/client/node_modules/@angular/common/esm5/http/testing.js:421:1)
	    at UserContext.<anonymous> Users/john/dev/swagger-angular-generator/demo-app/client/src/tests/post/order.service.spec.ts:29:17)
```

**vs. new**

```
OrderService should check request parameters are correct FAILED
	Expected object to have properties
	    attr: 16
	    at http://localhost:9876/_karma_webpack_/webpack:/Users/john/dev/swagger-angular-generator/demo-app/client/src/tests/post/order.service.spec.ts:33:28
```

Please consider it.